src/repositories/broker/proxies/freeWeatherAPI/FreeWeatherAPIProxy.ts

```diff
@@ -0,0 +1,27 @@
+import Proxy from '../Proxy';
+import WeatherResponse from '../../WeatherResponse';
+import FreeWeatherAPIService from '../../services/FreeWeatherAPIService';
+import FreeWeatherAPIResponseBuilder from './FreeWeatherAPIResponseBuilder';
+import { injectable } from 'tsyringe';
+
+@injectable()
+class FreeWeatherAPIProxy implements Proxy {
+    service: FreeWeatherAPIService;
+    responseBuilder: FreeWeatherAPIResponseBuilder;
+
+    constructor(service: FreeWeatherAPIService, responseBuilder: FreeWeatherAPIResponseBuilder) {
+        this.service = service;
+        this.responseBuilder = responseBuilder;
+    }
+
+    async getWeatherResponse(cityName: string): Promise<WeatherResponse> {
+        try {
+            const currentWetherResponse = await this.service.getCurrentWeather(cityName);
+            return this.responseBuilder.buildWeatherResponse(currentWetherResponse.data);
+        } catch (error) {
+            return this.responseBuilder.getEmptyReponse();
```

Comment:
Create a new case and extract this to a class, like class.getEmpty('service_name')